pdr_backend/lake/gql_data_factory.py

```diff
@@ -159,6 +162,37 @@ def _prepare_temp_table(self, table_name, st_ut, fin_ut, schema):
             data = CSVDataStore(table.base_path).read(table_name, st_ut, fin_ut, schema)
             table._append_to_db(data, TableType.TEMP)
 
+    @enforce_types
+    def _calc_start_ut_from_predictions(self) -> UnixTimeMs:
+        """
+        @description
+            Calculate start timestamp, reconciling whether file exists and where
+            its data starts. If file exists, you can only append to end.
+
+        @return
+            start_ut - timestamp (ut) to start grabbing data for (in ms)
+        """
+
+        # check if the predictions table exists in the database
+        if not PersistentDataStore(self.ppss.lake_ss.lake_dir).table_exists(
+            predictions_table_name
+        ):
+            return UnixTimeMs(self.ppss.lake_ss.st_timestamp + 1000)
```

Comment:
Why are we adding 1000 to the timestamp?

Reply:
It is unnecessary to add 1000 to the timestamp not retrieved from the DB. I agree

Reply:
removed